File: be/internal/ai/gender.go
package ai

import "strings"

// GenderPromptBlock hints how Hani addresses the user.
func GenderPromptBlock(gender string) string {
	switch strings.TrimSpace(gender) {
	case "male":
		return `[partner gender: male]
- He is your boyfriend/partner learning Korean
- You may use 오빠 when affection is high — not every message
- Do not feminize him or use 언니/누나 for him`
	case "female":
		return `[partner gender: female]
- She is your girlfriend/partner learning Korean
- You may use 언니 or 누나 when affection is high — pick one and stay consistent
- Do not use 오빠 for her`
	case "other":
		return `[partner gender: neutral / not specified]
- Use their name; stay gender-neutral in Korean honorifics
- Avoid assuming 오빠, 언니, or 누나 until they signal preference`
	default:
		return ""
	}
}

File: fe/lib/auth/gender.ts
export type UserGender = "male" | "female" | "other";

export const GENDER_OPTIONS: {
  id: UserGender;
  label: string;
  desc: string;
}[] = [
  { id: "male", label: "Nam", desc: "Hani gọi bạn kiểu 오빠" },
  { id: "female", label: "Nữ", desc: "Hani gọi bạn kiểu 언니/누나" },
  { id: "other", label: "Khác", desc: "Hani dùng tên bạn, trung tính" },
];
